File: greedy.c
#include <stdio.h>
#include <cs50.h>

void calculateCoins(int change);
int main(void){
    float input;
    int change;
    printf("O hai! ");
    
    // prompts user input
    do 
    {
     printf("How much change is owed?\n");
     input = get_float();
     change =  (long) ((input* 100)+0.5); 
    }
    while (input < 0);
    
    //calls calculateCoins function
    calculateCoins(change);
}

    // calculate change at end of each loop and count loops until change is 0 and prints
 void calculateCoins(int change)
{
    int coins = 0;
    int quarters = 25;
    int dimes = 10;
    int nickles = 5;
    int pennies = 1;
 
    while(change >=quarters)
{
        change = change - quarters;
        coins++;
    }
    while(change >=dimes)
{
        change = change - dimes;
        coins++;
    }
    while(change >=nickles)
{
        change = change - nickles;
        coins++;
    }
    while(change >=pennies)
{
        change = change - pennies;
        coins++;
    }
    
 printf("%i\n", coins);
}

File: greedy.py
import cs50

def main():
    """Prompt user input and call calculateCoins."""
    
    print("O hai! ", end="")
    
    # prompts user input
    while True:
        print("How much change is owed?")
        n = cs50.get_float()
        y = (n * 100)
        if n > 0.00:
            break
        
    # calls calculateCoins method
    calculateCoins(y)


def calculateCoins(y):
    """Calculate change at end of each loop and count loops until change is 0 and prints."""
    
    coins = 0
    while y >= 25:
        y = y - 25
        coins += 1

    while y >= 10:
        y = y - 10
        coins += 1

    while y >= 5:
        y = y - 5
        coins += 1

    while y >= 1:
        y = y - 1
        coins += 1

    print("{}".format(coins))
        

if __name__ == "__main__":
    main()
        
    